src/server/fedavg.py

```diff
@@ -218,6 +218,37 @@ def __init__(
         if use_fedavg_client_cls:
             self.init_trainer()
 
+        # create setup for centralized evaluation
+        self.dataset = self.get_dataset()
```

Comment:
`self.init_trainer()` will also introduce a dataset obejct. I think we should re-use the dataset object (and the indices as well) for avoiding unnecessary memory overhead.